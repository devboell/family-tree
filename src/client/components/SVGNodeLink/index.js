import React from 'react'
import PropTypes from 'prop-types'
import { isEmpty } from 'lodash/fp'

const SVGNodeLink = ({
  x,
  y,
  linkTo,
  dimensions,
}) => {
  const { boxWidth, marginY } = dimensions
  return (
    <g>
      {!isEmpty(linkTo)
        ? <path d={`M ${x + (boxWidth / 2)} ${y} V ${y - (marginY / 2)} H ${linkTo.x} V ${linkTo.y}`} />
        : null
      }
    </g>
  )
}

SVGNodeLink.propTypes = {
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  linkTo: PropTypes.shape({}), // eslint-disable-line
  dimensions: PropTypes.shape({}).isRequired,
}


export default SVGNodeLink
