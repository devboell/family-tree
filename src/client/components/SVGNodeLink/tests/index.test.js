import React from 'react'
import SVGNodeLink from '../index'

let props = {
  x: 100,
  y: 150,
  linkTo: {
    x: 200,
    y: 100,
  },
  dimensions: {
    boxWidth: 80,
    marginY: 50,
  },
}


it('SVGNodeLink with linkTo, renders component', () => {
  const wrapper = shallow(<SVGNodeLink {...props} />)
  expect(wrapper).toMatchSnapshot()
})

it('SVGNodeLink with linkTo, renders component', () => {
  props = {
    ...props,
    linkTo: {},
  }
  const wrapper = shallow(<SVGNodeLink {...props} />)
  expect(wrapper).toMatchSnapshot()
})
