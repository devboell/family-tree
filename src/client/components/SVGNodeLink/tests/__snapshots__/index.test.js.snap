// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGNodeLink with linkTo, renders component 1`] = `
<g>
  <path
    d="M 140 150 V 125 H 200 V 100"
  />
</g>
`;

exports[`SVGNodeLink with linkTo, renders component 2`] = `<g />`;
